jax/_src/numpy/lax_numpy.py

```diff
@@ -5639,6 +5618,78 @@ def take(
     indices_are_sorted: bool = False,
     fill_value: StaticScalar | None = None,
 ) -> Array:
+  """Take elements from an array.
+
+  JAX implementation of :func:`numpy.take`, implemented in terms of
+  :func:`jax.lax.gather`. JAX's behavior differs from NumPy in the case
+  of out-of-bound indices; see the ``mode`` parameter below.
+
+  Args:
+    a: array from which to take values.
+    indices: N-dimensional array of integer indices of values to take from the array.
+    axis: the axis along which to take values. If not specified, the array will
+      be flattened before indexing is applied.
+    mode: Out-of-bounds indexing mode, either ``"fill"`` or ``"clip"``. The default
+      ``mode="fill"`` returns invalid values (e.g. NaN) for out-of bounds indices;
+      the ``fill_value`` argument gives control over this value. For more discussion
+      of ``mode`` options, see :attr:`jax.numpy.ndarray.at`.
```

Comment:
OOC why does the parameter default to `None` instead of `"fill"`? Is this for compatibility with NumPy?

Reply:
No, NumPy doesn't have this parameter. It is basically plumbed through to `lax.gather`, which defaults to `None`, so it made sense to also default to `None` here.